### ISSUE TITLE
Provide optional token flag for private git repos.

### ISSUE THREAD

**Issue by hzlmn** [saojs/sao#74]
Hi thanks for great tool!

Currently it has not ability to load template from private repo. 
Would be nice to have something like `-t/--token`  for passing Github/etc. auth token.

If you consider it as an addition i could work on PR.

**Comment by egoist** [comment]
@hzlmn it can, simply use `--clone` flag if you can access it

**Comment by hzlmn** [comment]
Yes, it will work but there are few caveats. For example, if you've never accessed repo by ssh sao will freeze with such message, cuz it is not trusted yet.
```
The authenticity of host 'github.com (192.30.253.113)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
```

**Comment by egoist** [comment]
But that message will be gone for good once you accessed the host 😄  it's a lot simpler than the `token` solution.
